slider/replay.py

```diff
@@ -576,6 +636,62 @@ def parse(cls, data, *, library=None, client=None, save=False):
             **mod_kwargs,
         )
 
+    @lazyval
+    def hits(self):
+        b = self.beatmap
+        rdata = self.actions
+        scores = {"perfects": [],
+                  "goods": [],
+                  "bads": [],
+                  "misses": [],
+                  "sliderbreaks": [],
+                  }
+        hw = od_to_ms(b.od(easy=self.easy, hard_rock=self.hard_rock))
+        rad = circle_radius(b.cs(easy=self.easy, hard_rock=self.hard_rock))
+        i = 0
+        for obj in b.hit_objects:
+            if self.hard_rock:
```

Comment:
I guess it could return the pointer which gets forwarded to the next call. idk if that is cleaner. As far as moving things to statics they would be "private" like `_process_slider` but again, that may not be better. What do you think?